libs/net/vmspec.py

```
@@ -194,3 +202,90 @@ def wait_for_ifaces_status(
+    watch_from_resource_version = resource_version or vmi_instance.metadata.resourceVersion
+    for event in vmi.watcher(timeout=timeout, resource_version=watch_from_resource_version):
+        if event["type"] != "MODIFIED":
+            continue
+        if not _vmi_condition_present(
+            existing_conditions=event["object"].status.conditions or [], required_condition=condition, status=status
+        ):
+            return
+
+    raise VMIConditionStillPresentError(
+        f"VMI {vm.name}: condition {condition}={status} did not become missing or False within {timeout}s."
+    )
```

Ruff (0.15.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)